source/android/Additions/TraceWriter.cs

```diff
@@ -92,159 +89,13 @@ public static void WriteTrace(object exception)
-                                    foreach (Match m in _StackTraceLine.Matches(trace))
-                                    {
-                                        var method = m.Groups[1].Value;
-                                        if (AppNamespaces != null)
-                                        {
-                                            //use an application provided list of classes to determine which namespaces
-                                            //should be mapped into the package name so hockey app can pick out the correct
-                                            //top stacktrace line.  It unfortunately means you end up with an extra copy of
-                                            //package name at the beginning of your c# crash traces
-                                            foreach (var prefix in AppNamespaces)
-                                            {
-                                                if (method.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
-                                                {
-                                                    method = _AppPackage + "." + method;
-                                                    break;
-                                                }
-                                            }
-                                        }
-                                        else
-                                        {
-                                            //default to things that don't match mono/android/java, again, this forces an extra copy of
-                                            //package name at the beginning of your c# crash traces
-                                            if (!method.StartsWith("mono", StringComparison.OrdinalIgnoreCase) &&
-                                                !method.StartsWith("android", StringComparison.OrdinalIgnoreCase) &&
-                                                !method.StartsWith("system", StringComparison.OrdinalIgnoreCase) &&
-                                                !method.StartsWith("java", StringComparison.OrdinalIgnoreCase))
-                                            {
-                                                method = _AppPackage + "." + method;
-                                            }
-                                        }
-                                        //this forces the arguments part to look more like the file line number part that
-                                        //android stack traces normally have
-                                        var arguments = m.Groups[2].Value.Trim();
-                                        arguments = arguments.Replace(' ', '_');
-                                        arguments = arguments.Replace(',', '_');
+                ExceptionHandler.SaveException(Java.Lang.Throwable.FromException(exception as Exception), null, Java.Lang.Thread.CurrentThread(), _Listener);
```

Comment:
The Exceptions simply get converted to a JavaProxyThrowable that maintains the managed trace:

```
[HockeyApp] md52ce486a14f4bcd95899665e9d932190b.JavaProxyThrowable: TestHockeyApp.TestHockeyAppException: This is a basic exception.
[HockeyApp]   at TestHockeyApp.TestPage.ThrowBasicException (System.Object sender, System.EventArgs e) [0x00006] in /Users/michaeldang/Projects/TestHockeyApp/TestHockeyApp/TestPage.cs:83 
[HockeyApp]   at Xamarin.Forms.Button.Xamarin.Forms.IButtonController.SendClicked () [0x00020] in <filename unknown>:0 
[HockeyApp]   at Xamarin.Forms.Platform.Android.ButtonRenderer+ButtonClickListener.OnClick (Android.Views.View v) [0x00015] in <filename unknown>:0 
[HockeyApp]   at Android.Views.View+IOnClickListenerInvoker.n_OnClick_Landroid_view_View_ (IntPtr jnienv, IntPtr native__this, IntPtr native_v) [0x00011] in /Users/builder/data/lanes/3053/a94a03b5/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Android.Views.View.cs:1903 
[HockeyApp]   at (wrapper dynamic-method) System.Object:754b0014-0129-4993-8f1a-2ce0ac8dd019 (intptr,intptr,intptr)
[HockeyApp] 	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ButtonRenderer_ButtonClickListener.n_onClick(Native Method)
[HockeyApp] 	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ButtonRenderer_ButtonClickListener.onClick(ButtonRenderer_ButtonClickListener.java:29)
[HockeyApp] 	at android.view.View.performClick(View.java:4780)
[HockeyApp] 	at android.view.View$PerformClick.run(View.java:19866)
[HockeyApp] 	at android.os.Handler.handleCallback(Handler.java:739)
[HockeyApp] 	at android.os.Handler.dispatchMessage(Handler.java:95)
[HockeyApp] 	at android.os.Looper.loop(Looper.java:135)
[HockeyApp] 	at android.app.ActivityThread.main(ActivityThread.java:5254)
[HockeyApp] 	at java.lang.reflect.Method.invoke(Native Method)
[HockeyApp] 	at java.lang.reflect.Method.invoke(Method.java:372)
[HockeyApp] 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
[HockeyApp] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```